x/evm/keeper/state_transition.go

```diff
@@ -163,10 +166,10 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 
 	// Commit and switch to committed context
 	if !res.Failed() {
-		commit()
+		k.CommitCacheContexts()
 	}
 
-	k.EndCachedContext()
+	k.RevertToSnapshot(snapshot)
```

Comment:
It's weird to me that we "revert" even when the transaction succeeds... should we create a `Reset()` context function?

Reply:
agree, `revert` sounds like we are doing extra operation to return to previous state while we should just discard the cache (basically not commit)

Reply:
it's actually not needed, removed.